### PR TITLE
Implement tracer solution

### PR DESCRIPTION
**Pull request by atgeirr**
Initial implementation of tracer solve.

Not yet tested against other implementations.

### CODE REVIEW

**Comment by atgeirr** [comment]
@bska could you take a look at this, eventually merge?

**Comment by bska** [comment]
What is the "fixing" alluded to in commit 16bb697?

**Comment by atgeirr** [comment]
It was added as a debugging measure, the "fixing" is because the status at that point was that tracer was buggy. The "fixing" was accomplished by the next commit.

**Comment by bska** [comment]
> The "fixing" was accomplished by the next commit.

Okay, so the print messages could be turned into regular assertions?

**Comment by atgeirr** [comment]
> Okay, so the print messages could be turned into regular assertions?

I'll update the PR!

**Comment by atgeirr** [comment]
Update pushed, now local tof and tracer values are actually checked. Thanks for the catch!

File: tests/test_flowdiagnosticstool.cpp [hunk 1]
```diff
@@ -390,6 +394,7 @@ BOOST_AUTO_TEST_CASE (OneDimCase)
             BOOST_TEST_MESSAGE("[" << i << "] -> ToF["
                                << v.first << "] = "
                                << v.second);
+            BOOST_CHECK_EQUAL(v.second, expected[i]);
```
Comment (bska) [inline tests/test_flowdiagnosticstool.cpp]: Better make this `BOOST_CHECK_CLOSE` since this tests floating-point types.

**Comment by atgeirr** [comment]
> Better make this BOOST_CHECK_CLOSE since this tests floating-point types.

Good point, done.

**Comment by bska** [comment]
> > Better make this `BOOST_CHECK_CLOSE` since this tests floating-point types.
> 
> Good point, done.

Thanks a lot.  This looks good now and I'll merge into master.

**Pull request merged**